packages/core/src/lib/template-engine/nunjucksCompiler.ts

```diff
@@ -128,7 +128,10 @@ export class NunjucksCompiler implements Compiler {
     } else if (extension instanceof FilterRunner) {
       this.runtimeEnv.addFilter(
         extension.filterName,
-        extension.__transform.bind(extension),
+        function (this: any, value: any, ...args) {
+          // use classic function to receive context
+          extension.__transform(this, value, ...args);
+        },
```

Comment:
Nunjucks binding "context" to "this" (js this) https://github.com/mozilla/nunjucks/blob/master/nunjucks/src/compiler.js#L499

I'd want to pass context to transform functions, but we bound extension object to __transform function, it makes `this` become `extension` instead of the context provided by upstream.
```ts
extension.__transform.bind(extension),
```

So I used a classic function here the receive `this` first then pass the ___transform function as an argument.